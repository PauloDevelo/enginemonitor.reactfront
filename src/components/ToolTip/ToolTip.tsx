import React, { Fragment } from 'react';
import { UncontrolledTooltip } from 'reactstrap';
import { FormattedMessage } from 'react-intl';

import { faQuestion } from "@fortawesome/free-solid-svg-icons";
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import {UID} from 'react-uid';

import PropTypes from 'prop-types';

type ToolTipProps = {
	tooltip: FormattedMessage.MessageDescriptor
};

export default function ToolTip({tooltip}: ToolTipProps) {
	return (
		<UID>
			{id => (
				<Fragment>
					<span id={'tooltip' + id}>{' '}<FontAwesomeIcon icon={faQuestion} size="xs" color="grey"/></span>
					<UncontrolledTooltip target={'tooltip' + id}>
						<FormattedMessage {...tooltip} />
					</UncontrolledTooltip>
				</Fragment>
			)}
		</UID>
	);
}

ToolTip.propTypes = {
	tooltip: PropTypes.object.isRequired
};