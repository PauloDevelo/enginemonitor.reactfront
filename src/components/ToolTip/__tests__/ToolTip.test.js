import React from 'react';
import { UncontrolledTooltip } from 'reactstrap';
import { FormattedMessage } from 'react-intl';

import {  mount, shallow } from 'enzyme';

import ToolTip from '../ToolTip'

describe('Component ToolTip', () =>{
    let element;
    let container;

    beforeEach(() => {
        element = document.createElement('div');
        container = document.createElement('div');
        element.innerHTML = '<p id="target">This is the tooltip <span id="innerTarget">target</span>.</p>';
        element.setAttribute('id', 'testContainer');
        container.setAttribute('id', 'container');
        element.appendChild(container);
        document.body.appendChild(element);
    
        jest.useFakeTimers();
      });

    it('should render as a simple tooltip', () => {
        // Arrange
        const message = { id:"tooltip", defaultMessage: "hello world", description: "a tooltip"};

        // Act
        const wrapper = mount(<ToolTip tooltip={message}/>, { attachTo: container });

        // Assert
        const innerToolTip = wrapper.find(UncontrolledTooltip);
        const formatedMessage = innerToolTip.props().children;
    
        expect(formatedMessage.props.defaultMessage).toEqual(message.defaultMessage);
        expect(formatedMessage.props.id).toEqual(message.id);
    });

    it('should be the same as the previous snapshot', () => {
        // Arrange
        const message = { id:"tooltip", defaultMessage: "hello world", description: "a tooltip"};

        // Act
        const wrapper = mount(<ToolTip tooltip={message}/>, { attachTo: container });

        // Assert
        expect(wrapper).toMatchSnapshot();
    });

    
});