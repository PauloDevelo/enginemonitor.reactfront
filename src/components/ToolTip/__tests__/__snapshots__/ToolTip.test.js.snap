// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component ToolTip should be the same as the previous snapshot 1`] = `
<ToolTip
  tooltip={
    Object {
      "defaultMessage": "hello world",
      "description": "a tooltip",
      "id": "tooltip",
    }
  }
>
  <UID>
    <span
      id="tooltip2"
    >
       
      <FontAwesomeIcon
        border={false}
        className=""
        color="grey"
        fixedWidth={false}
        flip={null}
        icon={
          Object {
            "icon": Array [
              384,
              512,
              Array [],
              "f128",
              "M202.021 0C122.202 0 70.503 32.703 29.914 91.026c-7.363 10.58-5.093 25.086 5.178 32.874l43.138 32.709c10.373 7.865 25.132 6.026 33.253-4.148 25.049-31.381 43.63-49.449 82.757-49.449 30.764 0 68.816 19.799 68.816 49.631 0 22.552-18.617 34.134-48.993 51.164-35.423 19.86-82.299 44.576-82.299 106.405V320c0 13.255 10.745 24 24 24h72.471c13.255 0 24-10.745 24-24v-5.773c0-42.86 125.268-44.645 125.268-160.627C377.504 66.256 286.902 0 202.021 0zM192 373.459c-38.196 0-69.271 31.075-69.271 69.271 0 38.195 31.075 69.27 69.271 69.27s69.271-31.075 69.271-69.271-31.075-69.27-69.271-69.27z",
            ],
            "iconName": "question",
            "prefix": "fas",
          }
        }
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size="xs"
        spin={false}
        swapOpacity={false}
        symbol={false}
        title=""
        transform={null}
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-question fa-w-12 fa-xs "
          color="grey"
          data-icon="question"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 384 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M202.021 0C122.202 0 70.503 32.703 29.914 91.026c-7.363 10.58-5.093 25.086 5.178 32.874l43.138 32.709c10.373 7.865 25.132 6.026 33.253-4.148 25.049-31.381 43.63-49.449 82.757-49.449 30.764 0 68.816 19.799 68.816 49.631 0 22.552-18.617 34.134-48.993 51.164-35.423 19.86-82.299 44.576-82.299 106.405V320c0 13.255 10.745 24 24 24h72.471c13.255 0 24-10.745 24-24v-5.773c0-42.86 125.268-44.645 125.268-160.627C377.504 66.256 286.902 0 202.021 0zM192 373.459c-38.196 0-69.271 31.075-69.271 69.271 0 38.195 31.075 69.27 69.271 69.27s69.271-31.075 69.271-69.271-31.075-69.27-69.271-69.27z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </FontAwesomeIcon>
    </span>
    <UncontrolledTooltip
      target="tooltip2"
    >
      <Tooltip
        autohide={true}
        isOpen={false}
        placement="top"
        placementPrefix="bs-tooltip"
        target="tooltip2"
        toggle={[Function]}
        trigger="hover focus"
      >
        <TooltipPopoverWrapper
          autohide={true}
          delay={
            Object {
              "hide": 50,
              "show": 0,
            }
          }
          fade={true}
          hideArrow={false}
          innerClassName="tooltip-inner"
          isOpen={false}
          placement="top"
          placementPrefix="bs-tooltip"
          popperClassName="tooltip show"
          target="tooltip2"
          toggle={[Function]}
          trigger="hover focus"
        />
      </Tooltip>
    </UncontrolledTooltip>
  </UID>
</ToolTip>
`;
